011-php-code-formatting.md

```diff
@@ -0,0 +1,94 @@
+
+You can see a [pull request](https://github.com/DoSomething/rogue/pull/1094) on our Rogue application with how the Prettier PHP Plugin is configured and how it proceeded to format the code.
+
+This [commit](https://github.com/DoSomething/rogue/pull/1094/commits/5e2a94b46168539a6346b6057efc2d915e444626) specifically shows how the Prettier PHP plugin is set up to run along side the configuration for our JavaScript code.
+
+This [commit](https://github.com/DoSomething/rogue/pull/1094/commits/88a8ed7d2e5f8d3e13e93c4840b2d389cd331614) specifically shows all the files changed when the `npm run format` command was executed.
+
+The way the plugin is configured, in daily use developers will not need to manually run the command to format their code since it will run as a `pre-commit` hook on any changed files.
+
+Additionally, some code editors (like Visual Studio Code) easily support running Prettier on file saves (refer to the code editor docs for documentation on setup).
+
+### PHP Code Standards Fixer
+
+The [PHP Coding Standards Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) tool cares less about the "look" of the PHP Code, and more about the code quality. It focuses on automatically fixing PHP code to follow a specific set of code quality rules, and in some scenarios it can even modernize and optimize the code.
+
+It helps developers eliminate any concerns regarding the quality of their PHP code, whether they are adhering to [PSR](https://www.php-fig.org/psr/) standards and helps with linting their code.
+
+You can see a [pull request](https://github.com/DoSomething/rogue/pull/1092) on our Rogue application with how the PHP CS Fixer tool is configured and how it proceeded to format the code.
+
+This [commit](https://github.com/DoSomething/rogue/commit/eeca6efb6b29df01e13e816b7d051e872b8cd2f7) specifically shows how the PHP CS Fixer tool is set up. The process is simplified by utilizing the [weerd/php-style](https://github.com/weerd/php-style) package which contains easily accessible rulesets for both generic PHP packages and Laravel projects. This package is also version controlled, so developers can rest easy knowing they can specify a version of the ruleset to use and they can opt-in to ruleset changes by adjusting their composer package version. This scenario has been a pain point for the team in the past that affected us on [more than one](https://dosomething.slack.com/archives/C3ASB4204/p1572615964090600) [occasion](https://dosomething.slack.com/archives/CUQMU4Q6B/p1596116997002700?thread_ts=1596116997.002700), where StyleCI made a change and affected all our repositories since their rules are not version controlled.
+
+This [commit](https://github.com/DoSomething/rogue/pull/1092/commits/d48ec9a1a759941c49c5e687a80b639e9d808fed) specifically shows all the files changed when the `composer format` command was executed within the Homestead vagrant environment.
```

Comment:
Looks like mostly changing the order of import statements (from shortest to longest to...alphabetical?) and removing not needed spaces from string concatenation - are things things that Prettier can't handle?

Reply:
Relates to [this discussion](https://github.com/DoSomething/rfcs/pull/11#discussion_r469318473) with Dave. I'm going to enable Prettier first, then PHP CS Fixer after to compare what rules to remove from PHP CS Fixer that Prettier should only handle. There's some overlap but aiming to have the PHP linting happening via PHP CS Fixer which Prettier doesn't handle (thus the comparison to having both Prettier and ESLint in our JS). Here's a lot of what PHP CS Fixer will handle among PSR-2 rules: https://github.com/weerd/php-style/blob/master/src/rules/laravel.php

Reply:
Ah so is the PHP CS Fixer fully customizable? I only remember reading that Prettier is not

Reply:
Oh I could have been clearer on that. I mention the use of rulesets and changes being version controlled, but should maybe clarify that it is customizable. Ideally just once and you're done with it.

Reply:
Ah got it! I think that's why I was confused and thought we could get stuck in loops!

Reply:
@katiecrane I [updated the language](https://github.com/DoSomething/rfcs/pull/11/commits/fb3a4602d953c8803af6eb71615b2c0650f9913a) to help clarify this! Thanks for pointing it out!